tests/test_envs.py

```diff
@@ -1,142 +1,134 @@
 import pickle
 
 import gymnasium as gym
+import numpy as np
+import pytest
+from gymnasium.envs.registration import EnvSpec
 from gymnasium.utils.env_checker import check_env, data_equivalence
 
 import mo_gymnasium as mo_gym
-from mo_gymnasium import LinearReward
 
 
-def _test_pickle_env(env: gym.Env):
-    pickled_env = pickle.loads(pickle.dumps(env))
-
-    data_equivalence(env.reset(), pickled_env.reset())
-
-    action = env.action_space.sample()
-    data_equivalence(env.step(action), pickled_env.step(action))
-    env.close()
-    pickled_env.close()
-
-
-def test_deep_sea_treasure():
-    env = mo_gym.make("deep-sea-treasure-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_fishwood():
-    env = mo_gym.make("fishwood-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_four_room():
-    env = mo_gym.make("four-room-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_minecart():
-    env = mo_gym.make("minecart-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_mountaincar():
-    env = mo_gym.make("mo-mountaincar-v0")
-    env = LinearReward(env)
-    check_env(env, skip_render_check=True)
-    _test_pickle_env(env)
-
-
-def test_continuous_mountaincar():
-    env = mo_gym.make("mo-mountaincarcontinuous-v0")
-    env = LinearReward(env)
+all_testing_env_specs = []
+for env_spec in gym.envs.registry.values():
+    if type(env_spec.entry_point) is not str:
+        continue
+    # collect MO Gymnasium envs
+    if env_spec.entry_point.split(".")[0] == "mo_gymnasium":
+        # Ignore highway as they do not deal with the random seed appropriately
+        if not env_spec.id.startswith("mo-highway"):
```

Comment:
I think we should still implement tests for these guys (at least for the CI to tell us if they're broken)

Reply:
I saw on the highway-env repo and they fixed this already, but it is not released on pypi yet. So it should be fixed in their next release

Reply:
I will open an issue for this.